### ISSUE TITLE
Fails on Enums with lifetimes

### ISSUE THREAD

**Issue by CeleritasCelery** [paholg/subenum#5]
If I have the following code

```rust
use subenum::subenum;

#[subenum]
pub(crate) enum Enum<'ob> {
    #[subenum(Foo)]
    A(i64),
    B(&'ob i64),
}
```
it will fail to build with the following error

```
error[E0726]: implicit elided lifetime not allowed here
 --> src/main.rs:5:15
  |
5 |     #[subenum(Foo)]
  |               ^^^ expected lifetime parameter
  |
  = note: assuming a `'static` lifetime...

```

If I move the subenum to the other field

```rust
use subenum::subenum;

#[subenum]
pub(crate) enum Enum<'ob> {
    A(i64),
    #[subenum(Foo)]
    B(&'ob i64),
}
```
then it panic's with not implemented

```
error: custom attribute panicked
 --> src/main.rs:3:1
  |
3 | #[subenum]
  | ^^^^^^^^^^
  |
  = help: message: not implemented
```

**Comment by paholg** [comment]
Yeah, I didn't touch generics or lifetimes at all. I wanted to see if folks would actually find this useful before digging too deep. :)

I'll take a look tonight and see how hard this is to fix.

**Comment by CeleritasCelery** [comment]
I imagine this will be a little tricky, because the Enum may have multiple lifetimes and the subenum may only use some/none of them.

**Comment by paholg** [comment]
It's definitely doable; just need to filter the total generics by what's on all the variants. I'm just not sure when I'll have the time to do it.

**Comment by paholg** [comment]
I've just released version 1.0 with generic and lifetime support. Please let me know if it gives you any more problems.
